### PR TITLE
add maxLines prop to Text component

### PR DESCRIPTION
**Pull request by Morantron**
Taking https://github.com/recharts/recharts/issues/961#issuecomment-618265830 as inspiration, I've added a `maxLines` props to the `<Text />` component

image

# TODO

- [ ] Update docs

### CODE REVIEW

File: src/component/Text.tsx [hunk 1]
```diff
@@ -67,26 +68,56 @@ interface State {
...
+  if (!shouldLimitLines) {
     return result;
-  }, []);
+  }
+
+  let tempText = props.children as string;
+  const suffix = '…';
+
+  while (result.length > props.maxLines || findLongestLine(result).width > lineWidth) {
+    tempText = tempText.slice(0, -1);
```
Comment (xile611) [inline src/component/Text.tsx:110]: How about use binary search algorithm to find the right result?

Comment (Morantron) [reply src/component/Text.tsx:110]: good idea, i'll give a try :+1:

**Comment by Morantron** [comment]
Added binsearch approach. Might be cool in the future to extract all these helper functions to a TextUtils.ts file and test them independently. Maybe after https://github.com/recharts/recharts/pull/2380 is merged?

**Comment by avindra** [comment]
#2380 may take long (1-3mo if I am working on it alone). If you want to write the test now, I'll convert it later on. If you can test without enzyme that'd be :ok_hand: but if not, nbd

**Pull request merged**